apis/apiextensions/v1/composition_types.go

```diff
@@ -79,29 +49,15 @@ type CompositionSpec struct {
 	// +optional
 	// +listType=map
 	// +listMapKey=step
+	// +kubebuilder:validation:MinItems=1
+	// +kubebuilder:validation:MaxItems=99
```

Comment:
out of curiosity, where does this limit on pipeline steps come from?

Reply:
I just picked it arbitrarily, since I noticed you could theoretically write a very long pipeline and maybe cause Crossplane issues. To be fair 99 also seems very long, so maybe it should be even lower.